### PR TITLE
Added support for buckets usage (and more).

### PR DESCRIPTION
**Pull request by k0ste**
* Added support for buckets usage (in bytes).
* Fixed work with Ceph Luminous (#2).
* Added about 'admin api should be enabled'.
* Replace `-r` key to nagios standard `-H` for host definition.
* Added ability to change 'rgw admin entry' via `-e` key.
* Added ability to use https.
* Fix mistypes, delete extra/useless prefixes and symbols.

selection_001

### CODE REVIEW

File: radosgw_usage_exporter.py [hunk 1]
```diff
@@ -15,102 +16,177 @@
...
+                        category['bytes_sent'])
+
+                self._prometheus_metrics['bytes_received'].add_metric(
+                    [bucket_name, bucket_owner, category['category']],
+                        category['bytes_received'])
+
+
+    def _get_bucket_usage(self, bucket):
+        """
+        Metod get actual bucket usage (in bytes).
```
Comment (blemmenes) [inline radosgw_usage_exporter.py]: Did you mean "Method" here?

Comment (k0ste) [reply radosgw_usage_exporter.py]: Fixed.

**Comment by blemmenes** [comment]
First thanks for adding bucket support and all the general clean-up!

For me on hammer, the bucket portion you added doesn't work. For some reason before each dict with the bucket info they have a string with the bucket name. While the documentation between hammer and luminous doesn't mention the difference.

``` u'bjltest-04342-s3disk-16',
 {u'bucket': u'bjltest-04342-s3disk-16',
  u'bucket_quota': {u'enabled': False,
                    u'max_objects': -1,
                    u'max_size_kb': -1},
  u'id': u'default.10030135.31',
  u'index_pool': u'.rgw.buckets.index',
  u'marker': u'default.10030135.31',
  u'master_ver': u'0#0',
  u'max_marker': u'0#00000064941.79038.3',
  u'mtime': u'2017-10-26 14:05:33.000000',
  u'owner': u'softnas1',
  u'pool': u'.rgw.buckets',
  u'usage': {u'rgw.main': {u'num_objects': 4802,
                           u'size_kb': 4916225,
                           u'size_kb_actual': 4916228}},
  u'ver': u'0#64942'}
```



So here on 177 it breaks because that first element from the for loop in `collect` is a string not a dict.

https://github.com/k0ste/radosgw_usage_exporter/blob/e12960affcabe05f4bc258510f3d37ed9605934f/radosgw_usage_exporter.py#L167-L175

When I had started on bucket usage support I did something like this to iterate over every other item (starting with the second).

```python
if rgw_bucket:
    for bucket in rgw_bucket[1::2]:
        self._get_bucket_usage(bucket)
```

Not sure if the best approach would be to detect what the object type is inside the `if rgw_bucket` statement or detect the failure in `_get_bucket_usage`.

**Comment by k0ste** [comment]
@blemmenes can you paste full json response?

**Comment by blemmenes** [comment]
I can't give the full json as this is a production cluster, but here is several complete entries.

[hammer_bucket_dump.txt](https://github.com/blemmenes/radosgw_usage_exporter/files/1868472/hammer_bucket_dump.txt)

**Comment by k0ste** [comment]
Check me please:

```
radosgw_usage_bucket_total{bucket="a-08637-s3disk-41",owner="test",zonegroup="0"} 1048576.0
radosgw_usage_bucket_total{bucket="celab-rubrik-rubrik-4",owner="celab",zonegroup="0"} 8192.0
radosgw_usage_bucket_total{bucket="celab-rubrik-rubrik-1",owner="celab",zonegroup="0"} 8192.0
radosgw_usage_bucket_total{bucket="celab-rubrik-rubrik-0",owner="celab",zonegroup="0"} 12288.0
radosgw_usage_bucket_total{bucket="celab-rubrik-rubrik-3",owner="celab",zonegroup="0"} 8192.0
radosgw_usage_bucket_total{bucket="test",owner="blemmenes",zonegroup="0"} 8192.0
```

**Comment by blemmenes** [comment]
I'll try to take a look at this today.

**Comment by blemmenes** [comment]
@k0ste So this solution works great except for when a client creates a bucket that contains the word "bucket". Which of course I have. :)

A solution that worked for me was changing the if clause you added from `if 'bucket' in bucket:` to:

```python
if type(bucket) is dict:
    bucket_name = bucket['bucket']
    bucket_owner = bucket['owner'

<snip>
```

**Comment by k0ste** [comment]
@blemmenes updated.

**Comment by blemmenes** [comment]
@k0ste  Works great for me. Thanks for contributing!

**Pull request merged**